### DISCUSSION TITLE
No EC number in ec.eccodes although EC is in uniprot.tsv

### DISCUSSION THREAD

**Discussion by PkiwiBird** [SysBioChalmers/GECKO#390] [Q&A]
There's one reaction in a model that has an entry in `ec.enzymes`: 'A6NKP2' and it is present in `uniprot.tsv` file:
`A6NKP2	100288072	1.1.1.-	46869	MKSNPPRSSLEACKAAGQAPQQKTQAK`
but it is empty in ec.eccodes after running:
`model = getECfromDatabase(model)`
Then it gets a kcat value of 0 after running:
```
 kcatList_fuzzy  = fuzzyKcatMatching(model)
model  = selectKcatValue(model, kcatList_fuzzy)
```
There's also NaN in `.origin` of kcatList.
Could it be that the corresponding entry in ec.eccodes is empty because the EC, `1.1.1.-`, is not complete? and therefore does not find a kcat?
In that case, does it mean the fuzzy kcat matching only works after getting a complete EC? 
If so, would be interesting to get gecko to get the fuzzy kcat matching (for a similar enzyme, or same enzyme other substrate, etc) with an incomplete EC?

**Comment by edkerk** [comment]
The behaviour of `fuzzyKcatMatching` and `selectKcatValue` are as expected and intended, but `getECfromDatabase` should also fill incomplete EC codes such as `1.1.1.-`. In `full_ecModel/models/ecYeastGEM.yml` you can for instance find this example:

```yaml
  - !!omap
    - id: "r_2773"
    - kcat: 13.8
    - source: "brenda"
    - eccodes: "2.3.1.-"
    - enzymes: !!omap
        - Q06510: 1
```

So in your case it seems that there is something going wrong with finding the entry in `uniprot.tsv`, which is done by comparing the `ec.genes` field with the gene identifiers (2nd column in `uniprot.tsv`). That this fails is somewhat puzzling to me, as `makeEcModel` apparently did not have a problem to find the same gene, otherwise `A6NKP2` would never have ended up in `ec.enzymes`. Do you by any change use a `data/uniprotConversion.tsv` file? There might be a bit of difference how `getECfromDatabase` and `makeEcModel` deal with those, so maybe that is the issue.

Also, do you a `kegg.tsv` file? Because if the different handling of `data/uniprotConversion.tsv` is indeed the problem, then this likely means that the EC values that are ending up in your model are all derived from KEGG and not UniProt (in other words, the problem is not unique for this particular enzyme, but none of the enzymes get their EC from UniProt due to a bug in the mapping).
